espnet/espnet
# Split CI into several jobs

## What?

- Split CI into several jobs
	- `test_python.sh` -> `test_python_espnet1.sh` / `test_python_espnet2.sh`
	- `test_shell.sh` -> `test_shell_espnet1.sh` / `test_shell_espnet2.sh`
	- I found that the validation of configuraiton files in `test_integration_espnet2.sh` consume too much time so split it into `test_integration_espnet2.sh` and `test_configuration_espnet2.sh`.
- Changed OS / version matrix
	- MacOS / Windows: check only installation
	- Cent / Debian: check except integration test
	- Ubuntu
		- ESPnet1 related job: pytest / bats / integration of espnet1
			- Run on only oldest and newest combiniation (3.7 + 1.10.2  & 3.19 + 1.13.1)
		- ESPnet2 related job: pytest / bats / integration of espnet2
		    - Run on all supported pytorch with python 3.7
		    - Run on all python version with newest pytorch 1.13.1
		- ESPnet2 configuration validation job:	configuration check of espnet2
			- Run on only oldest and newest combiniation (3.7 + 1.10.2  & 3.19 + 1.13.1)

## Why?

- To reduce CI required time

## See also
- Should be merged after #5324 
- Potentially, we can finish CI within 1 hour including all steps (e.g., installation, pytest, integration test)
https://github.com/kan-bayashi/espnet/actions/runs/5629055287/job/15253593892?pr=9
